src/ALogOutputter.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ALogOutputter.hpp"
+
+namespace weblog {
+
+/**
+ * @brief Construct a new ALogOutputter::ALogOutputter object
+ *
+ * Default constructor for inheritance.
+ *
+ */
+ALogOutputter::ALogOutputter() { }
+
+/**
+ * @brief Destroy the ALogOutputter::ALogOutputter object
+ *
+ * Virtual destrcutor.
+ *
+ */
+ALogOutputter::~ALogOutputter() { }
+
+/**
+ * @brief PRIVATE copy constructor.
+ *
+ * This constructor is implemented to silence clang-tidy warnings.
+ *
+ * @param ref The reference object to copy.
+ */
+ALogOutputter::ALogOutputter(const ALogOutputter& ref) { (void)ref; }
+
+/**
+ * @brief PRIVATE copy assignment operator.
+ *
+ * This operator is implemented to silence clang-tidy warnings.
+ *
+ * @param rhs The right-hand side object to copy.
+ * @return ALogOutputter& The copied object.
+ */
+ALogOutputter& ALogOutputter::operator=(const ALogOutputter& rhs)
+{
+	if (this == &rhs)
```

Comment:
Just this should suffice:
```
(void)rhs;
return *this;
```

Reply:
No, because then you don't have a valid check for self assignment which gives a clang-tidy warning :)

Reply:
Can´t you mute clang-tidy for this? Seems like an unfitting warning to me.

Reply:
Added `//NOLINT` to silence warning, deleted check.